ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -159,6 +162,16 @@
               castExecutor(KsqlResource::distributeStatement, TerminateQuery.class))
           .build();
 
+  private static final Set<Class<? extends Statement>> SYNC_BLACKLIST =
```

Comment:
Added at @agavra's request in the previous PR: this is the set of non-distributed operations that do not require the app to wait for the last distributed operation to complete.

Reply:
If possible, I think we should prefer a whitelist to a blacklist. I am adding, for example, a custom handler for CSAS/CTAS statements and the test that was added is failing because now that it has a custom handler the "wait for previous" functionality is being resolved for them.

Reply:
A black list is preferred as new items should _by default_ wait on previous commands. This is the safer default. 

If we have a white list, then new entries added might not wait for the previous command, when they should..

Reply:
You have me convinced.